modules/ztp-creating-siteconfig-custom-resources.adoc

```diff
@@ -0,0 +1,351 @@
+
+|ClusterDeployment
+(`ClusterDeployment.yaml`)
+|References the AgentClusterInstall to use.
+|Used with InfraEnv to generate the Discovery ISO for the SNO cluster.
+
+|NMStateConfig
+(`NMStateConfig.yaml`)
+|Provides network configuration information such as `MAC` to `IP` mapping, DNS server, default route, and other network settings. This is not needed if DHCP is used.
+|Sets up a static IP address for the cluster’s Kube API server.
+
+|Agent
+|Contains hardware information about the target bare metal machine.
+|Created automatically on the hub when the target machine's Discovery ISO boots.
+
+|ManagedCluster
+(`ManagedCluster.yaml`)
+|When a cluster is managed by the hub, it must be imported and known. This Kubernetes object provides that interface.
+|The hub uses this resource to manage and show the status of clusters.
+
+|KlusterletAddonConfig
+(`KlusterletAddonConfig.yaml`)
+|Contains the list of services provided by the hub to be deployed to a ManagedCluster.
+|Tells the hub which addon services to deploy to a ManagedCluster.
+
+|Namespace
+(`namespace.yaml`)
+|Logical space for each ManagedCluster’s resources existing on the hub. Unique per site.
+|Propagates resources to the ManagedCluster.
+
+| Secret
+(`Secret.yaml`)
+(`PullSecret.yaml`)
+|Two CRs are created: BMC Secret and Image Pull Secret.
+a| * BMC Secret authenticates into the target bare metal machine using its username and password.
+* Image Pull Secret contains authentication information for the {product-title} image installed on the target bare metal machine.
+
+|ClusterImageSet
+|Contains {product-title} image information such as the repository and image name.
+|Passed into resources to provide {product-title} images.
+|===
+
+.Prerequisites
+
+* Enable Assisted Installer Service.
+
+* Ensure network connectivity:
+** The container within the hub must be able to reach the Baseboard Management Controller (BMC) address of the target bare metal machine.
+
+** The managed cluster must be able to resolve and reach the hub’s API `hostname` and `{asterisk}.app` host name.
+Example of the hub’s API and `{asterisk}.app` host name:
++
+[source,terminal]
+----
+console-openshift-console.apps.hub-cluster.internal.domain.com
+Api.hub-cluster.internal.domain.com
+----
+
+** The hub must be able to resolve and reach the API and `{asterisk}.app` host name of the managed cluster.
+Here is an example of the managed cluster’s API and `{asterisk}.app` host name:
++
+[source,terminal]
+----
+console-openshift-console.apps.sno-managed-cluster-1.internal.domain.com
+Api.sno-managed-cluster-1.internal.domain.com
+----
+
+** A DNS Server that is IP reachable from the target bare metal machine.
+
+* A target bare metal machine for the single node cluster with the following hardware minimums:
+
+** 4 CPU or 8 vCPU
+** 32 GiB RAM
+** 120 GiB Disk for root filesystem
+
+* Create a ClusterImageSet for each specific cluster version that needs to be deployed. A ClusterImageSet has the following format:
```

Comment:
> Prerequisites are conditions that must be satisfied before the user starts the procedure.

It's not really appropriate to include discrete steps like this in a prereqs section. See https://redhat-documentation.github.io/modular-docs/